File: self_library/src/main/java/com/wjf/self_library/view/codingUI/CommonToolbar.java
package com.wjf.self_library.view.codingUI;

import android.app.Activity;
import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.Color;
import android.util.AttributeSet;


import androidx.core.content.ContextCompat;

import com.wjf.self_library.R;
import com.wjf.self_library.common.BaseUi;
import com.wjf.self_library.databinding.UiCommonToolbarBinding;


/**
 * @author : Wangjf
 * @date : 2021/1/19
 */
public class CommonToolbar extends BaseUi<UiCommonToolbarBinding> {

    private String titleText, backText;
    private boolean darkTheme, ifBack;
    private int backgroundColor, backgroundChangeColor;

    public CommonToolbar(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    protected int setLayout() {
        return R.layout.ui_common_toolbar;
    }

    @Override
    protected int[] setStyleable() {
        return R.styleable.CommonToolbar;
    }

    @Override
    protected void getTypeArray(TypedArray typedArray) {
        titleText = typedArray.getString(R.styleable.CommonToolbar_titleText);
        backText = typedArray.getString(R.styleable.CommonToolbar_backText);
        darkTheme = typedArray.getBoolean(R.styleable.CommonToolbar_dark_theme, true);
        ifBack = typedArray.getBoolean(R.styleable.CommonToolbar_if_back, true);
        backgroundColor = typedArray.getColor(R.styleable.CommonToolbar_background_color, Color.TRANSPARENT);
        backgroundChangeColor = typedArray.getColor(R.styleable.CommonToolbar_background_change_color, Color.TRANSPARENT);
    }

    public int getBackgroundColor() {
        return backgroundColor;
    }

    public int getBackgroundChangeColor() {
        return backgroundChangeColor;
    }

    public void setTitleText(String titleText) {
        this.titleText = titleText;
        if (titleText != null) {
            view.toolbarTitle.setVisibility(VISIBLE);
            view.toolbarTitle.setText(titleText);
        } else {
            view.toolbarTitle.setVisibility(INVISIBLE);
        }
    }

    public boolean isDarkTheme() {
        return darkTheme;
    }

    @Override
    protected void setView() {
        int themeColor = darkTheme ? ContextCompat.getColor(context, R.color.black) : ContextCompat.getColor(context, R.color.white);

        if (ifBack) {
            view.backLine.setVisibility(VISIBLE);
            if (backText != null) {
                view.toolbarBack.setText(backText);
                view.toolbarBack.setVisibility(VISIBLE);
                view.toolbarBack.setTextColor(themeColor);
            }
            view.toolbarBackIcon.setVisibility(VISIBLE);
            view.toolbarBackIcon.changeColor(themeColor);
            view.backLine.setOnClickListener(v -> {
                if (toolbarBackCallback != null) {
                    toolbarBackCallback.backActivity(context);
                }
                ((Activity) context).finish();
            });
        }

        if (titleText != null) {
            view.toolbarTitle.setText(titleText);
            view.toolbarTitle.setVisibility(VISIBLE);
            view.toolbarTitle.setTextColor(themeColor);
        }

        setBackgroundColor(backgroundColor);
    }


    private ToolbarBackCallback toolbarBackCallback;


    public void setToolbarBackCallback(ToolbarBackCallback toolbarBackCallback) {
        this.toolbarBackCallback = toolbarBackCallback;
    }


    public interface ToolbarBackCallback {
        /**
         * 返回事件回调
         *
         * @param context 上下文对象
         */
        void backActivity(Context context);
    }
}

File: corekit/src/main/java/org/jxxy/debug/widget/CommonToolbar.kt
package org.jxxy.debug.widget

import android.app.Activity
import android.content.Context
import android.content.res.TypedArray
import android.graphics.Color
import android.util.AttributeSet
import android.view.LayoutInflater
import androidx.annotation.ColorInt
import org.jxxy.debug.R
import org.jxxy.debug.common.BaseUi
import org.jxxy.debug.databinding.UiCommonToolbarBinding
import org.jxxy.debug.util.*

/**
 * @author : Wangjf
 * @date : 2021/1/19
 */
class CommonToolbar(context: Context, attrs: AttributeSet?) : BaseUi<UiCommonToolbarBinding>(
    context, attrs
) {
    private var titleText: String? = null
    private var backText: String? = null
    var isDarkTheme = false
        private set
    private var ifBack = false

    @ColorInt
    var backgroundNormalColor = 0
        private set

    @ColorInt
    var backgroundChangeColor = 0
        private set

    var toolbarBackCallback: ToolbarBackCallback? = null

    override fun bindLayout(inflater: LayoutInflater) {
        view = UiCommonToolbarBinding.inflate(
            inflater,
            this,
            true
        )
    }

    override fun setStyleable(): IntArray {
        return R.styleable.CommonToolbar
    }

    override fun getTypeArray(typedArray: TypedArray) {
        titleText = typedArray.getString(R.styleable.CommonToolbar_titleText)
        backText = typedArray.getString(R.styleable.CommonToolbar_backText)
        isDarkTheme = typedArray.getBoolean(R.styleable.CommonToolbar_dark_theme, true)
        ifBack = typedArray.getBoolean(R.styleable.CommonToolbar_if_back, true)
        backgroundNormalColor =
            typedArray.getColor(R.styleable.CommonToolbar_background_color, Color.TRANSPARENT)
        backgroundChangeColor = typedArray.getColor(
            R.styleable.CommonToolbar_background_change_color, Color.TRANSPARENT
        )
    }

    fun setTitleText(titleText: String?) {
        this.titleText = titleText
        if (this.titleText.isNullOrEmpty()) {
            view.toolbarTitle.hide()
        } else {
            view.toolbarTitle.show()
            view.toolbarTitle.text = titleText
        }
    }

    override fun setView() {
        val themeColor =
            if (isDarkTheme) ResourceUtil.getColor(R.color.black)
            else ResourceUtil.getColor(R.color.white)

        if (!ifBack) {
            view.backLine.gone()
        } else {
            view.backLine.show()
            view.backLine.singleClick {
                toolbarBackCallback?.run {
                    backActivity(context)
                    return@singleClick
                }
                (context as? Activity)?.finish()
            }
            if (backText.isNullOrEmpty()) {
                view.toolbarBack.gone()
            } else {
                view.toolbarBack.show()
                view.toolbarBack.text = backText
                view.toolbarBack.setTextColor(themeColor)
            }
            view.toolbarBackIcon.show()
            view.toolbarBackIcon.setTextColor(themeColor)
        }

        setTitleText(titleText)
        view.toolbarTitle.setTextColor(themeColor)

        setBackgroundColor(backgroundNormalColor)
    }

    interface ToolbarBackCallback {
        /**
         * 返回事件回调
         *
         * @param context 上下文对象
         */
        fun backActivity(context: Context?)
    }
}
